lib/logstash/filters/mutate.rb

```diff
@@ -320,21 +324,32 @@ def convert_string(value)
     value.to_s.force_encoding(Encoding::UTF_8)
   end
 
+  def convert_boolean(value)
+    return true if value =~ TRUE_REGEX
+    return false if value.empty? || value =~ FALSE_REGEX
+    @logger.warn("Failed to convert #{value} into boolean.")
+    value
+  end
+
   def convert_integer(value)
     return 1 if value == true
     return 0 if value == false
-    value.to_i
+    value.squeeze(MUC_CP).tr(MUC_CP, MUC_UP).to_i
   end
 
   def convert_float(value)
-    value.to_f
+    value.squeeze(MUC_CP).tr(MUC_CP, MUC_UP).to_f
   end
 
-  def convert_boolean(value)
-    return true if value =~ TRUE_REGEX
-    return false if value.empty? || value =~ FALSE_REGEX
-    @logger.warn("Failed to convert #{value} into boolean.")
-    value
+  def convert_integer_eu(value)
+    return 1 if value == true
+    return 0 if value == false
```

Comment:
is converting to 0 and 1 from false and true a documented/historical behaviour or a new feature introduced here?

Reply:
nvmd - I see that `convert_integer` was already doing that.

Reply:
Thoughts on using [java.text.NumberFormat](https://docs.oracle.com/javase/8/docs/api/java/text/NumberFormat.html).[parse](https://docs.oracle.com/javase/8/docs/api/java/text/NumberFormat.html)() which supports Locales and whatnot?

Reply:
its as above for `convert_integer`

Reply:
I checked to see if JRuby used the Java text NumberFormatter but it does not. I presume to follow what CRuby does. I am concerned that NumberFormat **may** introduce new behaviour after reading this https://www.ibm.com/developerworks/library/j-numberformat/